### ISSUE TITLE
routeRules are not taken into account for dynamic url sources

### ISSUE THREAD

**Issue by tmlmt** [harlan-zw/nuxt-simple-sitemap#163]
### Describe the bug

In an i18n case, with this example of routeRules: 
```
   "/admin/**": { index: false }
```

The /admin pages still end up in the sitemaps

### Reproduction

https://stackblitz.com/edit/github-vty3pl-txnly4?file=nuxt.config.ts

### System / Nuxt Info

```Shell
------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.8.1
- CLI Version:      3.9.1
- Nitro Version:    2.7.2
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      devtools, modules, content, i18n, routeRules, site, sitemap
- Runtime Modules:  @nuxtjs/i18n@8.0.0-rc.5, @nuxt/content@2.8.5, nuxt-simple-sitemap@4.0.1
- Build Modules:    -
------------------------------
```

**Comment by harlan-zw** [comment]
Thanks, fixed in v4.1

**Comment by tmlmt** [comment]
Also works with i18n, meaning that setting `/admin/**`: {index: false}` removes all `/<locale>/admin/**` from the sitemaps. Great ! Thanks
